### PR TITLE
describe validationData, authorizer

### PR DESCRIPTION
**Pull request by drortirosh**

### CODE REVIEW

File: RIPS/rip-7560.md [hunk 1]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
 
 The frame must return 32 bytes `validationData` that is interpreted as:
 
-```solidity
-MAGIC_VALUE_SENDER <20 bytes>, validUntil <6 bytes>, validAfter <6 bytes>
-```
+- **authorizer** (20 bytes)
```
Comment (drortirosh) [inline RIPS/rip-7560.md]:
```suggestion
- **authorizer** (20 bytes) - (magic value/address, defined below)
```

File: RIPS/rip-7560.md [hunk 2]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
 [Calculation of Transaction Type AA_TX_TYPE hash](#calculation-of-transaction-type-aatxtype-hash).
 
-The frame must return a bytes array that is interpreted as:
+The frame returns 
 
-```solidity
-MAGIC_VALUE_PAYMASTER <20 bytes>, validUntil <6 bytes>, validAfter <6 bytes>, 64 <constant, 32 bytes>, length <32 bytes>, context <variable>
-```
-(this is the same encoding the the above solidity method returns)
+- `validationData` encoded as:
```
Comment (shahafn) [inline RIPS/rip-7560.md]: This is a duplicate, if `validationData` is identical, we can refer to the previous definition

File: RIPS/rip-7560.md [hunk 3]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
```
Comment (drortirosh) [inline RIPS/rip-7560.md]:
```suggestion
- `validationData` - encoded the same as described above, in validateTransaction.
```

File: RIPS/rip-7560.md [hunk 4]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
+- **authorizer** (20 bytes) - (magic value/address, defined below)
+- **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
+- **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
 
-In order to allow a gas estimation of the transaction prior signing, this function
-should avoid reverting on invalid signature, and return `MAGIC_VALUE_SIGFAIL` instead.
+Currently, **authorizer** is treated as a magic value:
+- `MAGIC_VALUE_SENDER` - returned by the account after it verified the transaction (and agrees to pay for its execution)
+- `MAGIC_VALUE_SIGFAIL` - returned by the account if the transaction is syntactically valid, but the signature is not.
+    This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.
```
Comment (forshtat) [inline RIPS/rip-7560.md]: This *value*

File: RIPS/rip-7560.md [hunk 5]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
 
 The frame must return 32 bytes `validationData` that is interpreted as:
 
-```solidity
-MAGIC_VALUE_SENDER <20 bytes>, validUntil <6 bytes>, validAfter <6 bytes>
-```
+- **authorizer** (20 bytes) - (magic value/address, defined below)
```
Comment (forshtat) [inline RIPS/rip-7560.md]: Don't introduce the new term "authorizer" in this EIP. Just make it 20 bytes. We can add the explanation in the "Rationale" section.

File: RIPS/rip-7560.md [hunk 6]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
+- **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
+- **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
 
-In order to allow a gas estimation of the transaction prior signing, this function
-should avoid reverting on invalid signature, and return `MAGIC_VALUE_SIGFAIL` instead.
+Currently, **authorizer** is treated as a magic value:
+- `MAGIC_VALUE_SENDER` - returned by the account after it verified the transaction (and agrees to pay for its execution)
```
Comment (forshtat) [inline RIPS/rip-7560.md]: Remove the brackets "(" and ")". They clutter the text and add no value.

File: RIPS/rip-7560.md [hunk 7]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
+- `MAGIC_VALUE_SIGFAIL` - returned by the account if the transaction is syntactically valid, but the signature is not.
+    This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.
+- Any other returned value is considered failure, and the transaction is rejected.
+- Future protocol MAY define authorizer as an on-chain contract to perform further account authorization, such as aggregation
```
Comment (forshtat) [inline RIPS/rip-7560.md]: Move this to rationale

File: RIPS/rip-7560.md [hunk 8]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
+- **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
+- **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
 
-In order to allow a gas estimation of the transaction prior signing, this function
-should avoid reverting on invalid signature, and return `MAGIC_VALUE_SIGFAIL` instead.
+Currently, **authorizer** is treated as a magic value:
+- `MAGIC_VALUE_SENDER` - returned by the account after it verified the transaction (and agrees to pay for its execution)
+- `MAGIC_VALUE_SIGFAIL` - returned by the account if the transaction is syntactically valid, but the signature is not.
+    This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.
+- Any other returned value is considered failure, and the transaction is rejected.
```
Comment (forshtat) [inline RIPS/rip-7560.md]: ... considered **to be a** failure

File: RIPS/rip-7560.md [hunk 9]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
+- Future protocol MAY define authorizer as an on-chain contract to perform further account authorization, such as aggregation
 
-Type of the `validUntil` is 6-byte timestamp value, or zero for "infinite". The transaction is valid only up to this time.
-Type of the `validAfter` is 6-byte timestamp. The transaction is valid only after this time.
-
-The `validateTransaction` function can choose to revert on any condition that can be satisfied during gas estimation.
+A revert by `validateTransaction` indicates rejection of the transaction, and it is not included on-chain.
```
Comment (forshtat) [inline RIPS/rip-7560.md]: Revert and invalid return value are equivalent, so they should be in the same element in the list.

File: RIPS/rip-7560.md [hunk 10]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
 
 ```
 
 The gas limit of this frame is set to `paymasterValidationGasLimit`.
 
 The amount of gas used by this frame is referred to as `paymasterValidationGasUsed`.
 
-The `transaction` parameter is interpreted as an ABI encoding of `TransactionType4`.\
-The `txHash` parameter represents the hash of the AA_TX_TYPE transaction with empty signature, as defined in section
+- The `version` parameter is `AA_TX_TYPE`
```
Comment (forshtat) [inline RIPS/rip-7560.md]: No, it isn't. The `version` is a mechanism to change the structure without necessarily changing the transaction type. This mechanism is still TBD though.

File: RIPS/rip-7560.md [hunk 11]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
-```solidity
-MAGIC_VALUE_PAYMASTER <20 bytes>, validUntil <6 bytes>, validAfter <6 bytes>, 64 <constant, 32 bytes>, length <32 bytes>, context <variable>
-```
-(this is the same encoding the the above solidity method returns)
+- `validationData` - encoded the same as described above, in validateTransaction.
+    - **authorizer** (20 bytes)
+    - **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
+    - **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
+- `context` - optional buffer returned by the paymaster, which is later passed to `postPaymasterTransaction`
+    This value MUST NOT exceed `MAX_CONTEXT_SIZE`.
```
Comment (forshtat) [inline RIPS/rip-7560.md:280]: Let's define `MAX_CONTEXT_SIZE` somewhere?

File: RIPS/rip-7560.md [hunk 12]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
 
-Same as in the [`sender` validation frame](#sender-validation-frame), in order to support gas estimation this
-frame should revert on errors, but return `MAGIC_VALUE_SIGFAIL` if the failure is because of signing error.
+The  **authorizer** is treated as a magic value:
+- `MAGIC_VALUE_PAYMASTER` - returned by the paymaster after it verified the transaction (and agrees to pay for its execution)
+- `MAGIC_VALUE_SIGFAIL` - returned by the paymaster if its paymasterData is expected to contain a signature, but its current value is not valid.
+    This valid is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.
```
Comment (forshtat) [inline RIPS/rip-7560.md]: This **value**

File: RIPS/rip-7560.md [hunk 13]
```diff
@@ -292,7 +300,7 @@ Calculation of the `calldataGasUsed` value is defined in the
 [Gas fees charged for transaction input](#gas-fees-charged-for-transaction-input) section.\
 The amount of gas used by this frame is referred to as `gasUsedByExecution`.
 
-The validation frames do not revert if the execution frame reverts.
+The transaction is included in the block even if the execution frame reverts.
```
Comment (forshtat) [inline RIPS/rip-7560.md]: I think my version is more explicit.

File: RIPS/rip-7560.md (hunk 5 above)
Comment (drortirosh) [reply RIPS/rip-7560.md]: then how to call it? 20 byte magic?

File: RIPS/rip-7560.md (hunk 6 above)
Comment (drortirosh) [reply RIPS/rip-7560.md]: not sure to which brakets you mean to remove.

File: RIPS/rip-7560.md [hunk 14]
```diff
@@ -241,47 +242,54 @@ The amount of gas used by this frame is referred to as `senderValidationGasUsed`
...
-```
+- **authorizer** (20 bytes) - (magic value/address, defined below)
+- **validUntil** (6 bytes) - a timestamp, or zero for "infinite". The transaction is valid only up to this time.
+- **validAfter** (6 bytes) - a timestamp. The transaction is valid only after this time.
 
-In order to allow a gas estimation of the transaction prior signing, this function
-should avoid reverting on invalid signature, and return `MAGIC_VALUE_SIGFAIL` instead.
+Currently, **authorizer** is treated as a magic value:
+- `MAGIC_VALUE_SENDER` - returned by the account after it verified the transaction (and agrees to pay for its execution)
+- `MAGIC_VALUE_SIGFAIL` - returned by the account if the transaction is syntactically valid, but the signature is not.
```
Comment (drortirosh) [inline RIPS/rip-7560.md]:
```suggestion
- `MAGIC_VALUE_SIGFAIL` - This value is returned by the account if the transaction is syntactically valid, but the signer is incorrect.
```

File: RIPS/rip-7560.md (hunk 8 above)
Comment (drortirosh) [inline RIPS/rip-7560.md]:
```suggestion
- Any other returned value is considered to be a failure, and the transaction is rejected.
```

File: RIPS/rip-7560.md (hunk 10 above)
Comment (drortirosh) [reply RIPS/rip-7560.md]: then suggest a fix to this line - either remove it altogether, or suggest another value (e.g. AA_TX_VERSION?)

File: RIPS/rip-7560.md (hunk 12 above)
Comment (drortirosh) [inline RIPS/rip-7560.md]:
```suggestion
    This value is used during gas estimation, and it helps wallet and apps to determine the gas used by the validation process.
```

File: RIPS/rip-7560.md (hunk 13 above)
Comment (drortirosh) [inline RIPS/rip-7560.md]:
```suggestion
The validation frames do not revert even if the execution frame reverts.
```

File: RIPS/rip-7560.md (hunk 11 above)
Comment (drortirosh) [reply RIPS/rip-7560.md:280]: defined in the constants section

**Review by forshtat — APPROVED** [review]

**Comment by forshtat** [comment]
@drortirosh note that we must make this PR also to the public repo here: https://github.com/ethereum/RIPs

**Pull request merged**